WordPress/WordPressShareExtension/ShareExtensionAbstractViewController.swift

```diff
@@ -18,6 +18,25 @@ class ShareExtensionAbstractViewController: UIViewController, ShareSegueHandler
 
     typealias CompletionBlock = () -> Void
 
+    // MARK: - Cache
+
+    internal static let cache = NSCache<AnyObject, AnyObject>()
+    static func clearCache() {
+        cache.removeAllObjects()
+    }
+    internal static func storeCategories(_ categories: [RemotePostCategory], for siteID: NSNumber) {
+        cache.setObject(categories as AnyObject, forKey: "\(siteID)-categories" as AnyObject)
+    }
+    internal static func cachedCategoriesForSite(_ siteID: NSNumber) -> [RemotePostCategory]? {
+        return cache.object(forKey: "\(siteID)-categories" as AnyObject) as? [RemotePostCategory]
+    }
+    internal static func storeDefaultCategoryID(_ defaultCategoryID: NSNumber, for siteID: NSNumber) {
+        cache.setObject(defaultCategoryID as AnyObject, forKey: "\(siteID)-default-category" as AnyObject)
```

Comment:
I know it's only a few short methods, but might be nice to have a small helper to generate the key string for a given site ID for both categories and default category (to save using the literal and `as AnyObject` in two places)